controllers/childGuidance.js

```diff
@@ -0,0 +1,154 @@
+import childModel from '../models/childModel.js';
+
+const createGuidance = async (req, res) => {
+  try {
+    const data = req.body;
+
+    const guideInstance = new childModel({
+      name: data.name,
+      age: data.age,
+      guides: [
+        {
+          question: guides.question,
+          answer:guides.answer,
+          category: guides.category,
+        },
+      ],
+    });
+
+    const result = await guideInstance.save();
+
+    res.status(200).json({
+      message: "Child Guidance created successfully",
+      error: null,
+      data: result,
+    });
+  } catch (error) {
+    res.status(500).json({
+      message: "Error in the createGuidance controller",
+      error: error,
+      data: null,
+    });
+  }
+}
+
+
+
+
+const getGuidance = async (req, res) => {
+  try {
+    const guideId = req.params.guideId;
+    const check = await childModel.findOne({ guideId: guideId });
+    if (!check) {
+      return res.status(409).json({
+        message: "The child Guide you are trying to read is not available",
+        error: "Error fetching the Child Guide",
+        data: null,
+      });
+    }
+    res.status(200).json({
+      message: "The child Guide fetched successfully",
+      error: null,
+      data: check,
+    });
+  } catch (error) {
+    res.status(500).json({
+      message: "Error in the getGuidance controller",
+      error: error,
+      data: null,
+    });
+  }
+};
+
+
+
+
+const getAllGuidances= async(req,res)=>{
+ try{
+  const page= req.query.page
+  const range= req.query.limit
+
+  const end= limit * page
+  const start= end - limit
+  const result= await childModel.find({}).skip(start).limit(range)
+  res.status(200).json({
+    message: "Data fetched successfully",
+    error: null,
+    data: response
+ });
+}
+
+catch(error){
+  console.log("Error in the getAllGuidance")
+  res.status(500).json({
+      message: "Failed to get childGuidances",
+      error: error,
+      data:null
+  })
+}
+
+}
+
+
+
+const updateGuidance = async (req, res) => {
+    try {
+      const data = req.body;
+      const guideId = req.params.guideId;
+  
+      const check = await childModel.find({guideId : guideId});
```

Comment:
- also here the find method will take `_id:guideId`

Reply:
I'm going to change it asap